ti/instrumentation/instrumentation.go

```diff
@@ -212,6 +230,20 @@ func GetCmd(ctx context.Context, config *api.RunTestConfig, stepID, workspace st
 	}
 
 	testCmd, err := runner.GetCmd(ctx, selection.Tests, config.Args, workspace, iniFilePath, artifactDir, cfg.GetIgnoreInstr(), !config.RunOnlySelectedTests)
+
+	if len(modules) != 0 && strings.Contains(testCmd, "bazel") {
```

Comment:
Move this whole logic to `GetCmd` of bazel.go
```suggestion
	if tiConfigYaml.Config.BazelOptimization & config.BuildTool == "bazel" {
	   testCmd = getBazelTestCmd(...)
	}
```